docs/src/proposals/program-recovery.md

```diff
@@ -0,0 +1,89 @@
+# Program Recovery
+
+## Problem
+
+As of Solana v1.11, the Upgradable Program Loader makes closing programs an irreversible action.
+
+After closing a program, the [`Program`] points to a non-existent [`ProgramData`] account.
+Note that the `Program` account itself still exists, as it is executable, and thus immutable.
+
+  [`Program`]: https://docs.rs/solana-sdk/latest/solana_sdk/bpf_loader_upgradeable/enum.UpgradeableLoaderState.html#variant.Program
+  [`ProgramData`]: https://docs.rs/solana-sdk/latest/solana_sdk/bpf_loader_upgradeable/enum.UpgradeableLoaderState.html#variant.ProgramData
+
+Below is an instance of a closed program (as of epoch 343).
+
+```
+# Irrecoverable Program account
+% solana account optFiKjQpoQ3PvacwnFWaPUAqXCETMJSz2sz8HwPe9B
+
+Public Key: optFiKjQpoQ3PvacwnFWaPUAqXCETMJSz2sz8HwPe9B
+Balance: 0.00114144 SOL
+Owner: BPFLoaderUpgradeab1e11111111111111111111111
+Executable: true
+Rent Epoch: 310
+Length: 36 (0x24) bytes
+0000:   02 00 00 00  89 d1 af 82  0a c8 fc e8  89 1f 84 d6   ................
+0010:   ee 7d 70 aa  34 60 99 85  fa 30 3a 08  37 ff b6 4d   .}p.4`...0:.7..M
+0020:   b7 20 60 c2                                          . `.
+
+# Irrecoverable ProgramData account
+% solana account AGzJVXAdQkyN8mDw619DuAnMyz4ypBJNZHSY7zKKJuvh
+Error: AccountNotFound: pubkey=AGzJVXAdQkyN8mDw619DuAnMyz4ypBJNZHSY7zKKJuvh
+```
+
+In Solana v1.11, the upgradable loader lacks instructions to recreate a `ProgramData` account that has been closed.
+Closed programs are thus permanently irrecoverable.
+Any PDAs associated with a closed program would be permanently inaccessible.
+
+This behavior is counterintuitive.
+Closed accounts of most other programs can be recreated (e.g. system program, SPL token program)
```

Comment:
I think that SPL token accounts are a really good example of why making this change is actually more intuitive since they can be recreated even if a new authority closes the account. Furthermore, most people don't realize that closing a program is irreversible which leads me to agree that it's an unintuitive implementation